dracut/10diskless-generator/diskless-generator

```
@@ -73,16 +73,28 @@ EOF
 
 # Alternatively support using btrfs in ram instead of tmpfs
 elif [[ -z "${root}" && "${rootfstype}" == btrfs ]]; then
-    add_requires diskless-btrfs.service
+    add_requires sysroot.mount
+    cat >"${UNIT_DIR}/sysroot.mount" <<EOF
+# Automatically generated by diskless-generator
+
+[Unit]
+Requires=diskless-btrfs.service
+After=diskless-btrfs.service
+Before=initrd-root-fs.target
+ConditionPathExists=/root.btrfs
```

This condition would allow this unit to be skipped if /root.btrfs fails to exist for some reason. Just drop it and let the unit fail if such an issue occurs.